### DISCUSSION TITLE
Display value / Berry script

### DISCUSSION THREAD

**Discussion by sheritafita** [arendst/Tasmota#16317] [Q&A] [answered]
Hello community,

I would like to display on an LCD the value of an analog input but it shows %value% instead of the real value.

Here is my code (Berry script) : 
```
def test_adc(value)
  tasmota.cmd("DisplayText Analog : %value% ")
end
tasmota.add_rule("ANALOG#A1",test_adc)
```
Any suggestions what should I do?
Thanks in advance

**Comment by sfromis** [comment, marked as answer]
The notation with %value% only works in classic rules, not in Berry.

What you can do in Berry is simplest `tasmota.cmd("DisplayText Analog : " + str(value))`

Or more general, for better formatting options:
```
import string
tasmota.cmd(string.format("DisplayText Analog : %d ", value))
```
This assumes an integer, you also have other formatting options, including decimals or floating-point. This way you can have a string with several placeholders and values.

**Reply by sheritafita** [reply]
It works, thank you so much for your detailed explanation !
